drivers/platform/linux/linux_gpio.c

```diff
@@ -73,34 +89,59 @@ int32_t gpio_get(struct gpio_desc **desc,
 	if (!descriptor)
 		return FAILURE;
 
+	linux_desc = (struct linux_gpio_desc*) malloc(sizeof(struct linux_gpio_desc));
+	if (!linux_desc)
+		goto free_desc;
+
+	descriptor->extra = linux_desc;
 	descriptor->number = param->number;
 
 	fd = open("/sys/class/gpio/export", O_WRONLY);
 	if (fd < 0) {
 		printf("%s: Can't open device\n\r", __func__);
-		free(descriptor);
-		return FAILURE;
+		goto free;
```

Comment:
nitpick: maybe an other label name can be used to not be the same as the function name.